llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -54605,7 +54610,132 @@ static bool onlyZeroFlagUsed(SDValue Flags) {
   return true;
 }
 
+static SDValue combineX86SubCmpToCcmpCtest(SDNode *N, SDValue Flag,
+                                           SelectionDAG &DAG,
+                                           TargetLowering::DAGCombinerInfo &DCI,
+                                           const X86Subtarget &ST) {
+  // cmp(and/or(setcc(cc0, flag0), setcc(cc1, sub (X, Y))), 0)
+  // brcond ne
+  //
+  //  ->
+  //
+  // ccmp(X, Y, cflags/~cflags, cc0/~cc0, flag0)
+  // brcond cc1
+  //
+  //
+  // sub(and/or(setcc(cc0, flag0), setcc(cc1, sub (X, Y))), 1)
+  // brcond ne
+  //
+  // ->
+  //
+  // ccmp(X, Y, cflags/~cflags, cc0/~cc0, flag0)
+  // brcond ~cc1
+
+  // cmp(and/or(setcc(cc0, flag0), setcc(cc1, cmp (X, 0))), 0)
+  // brcond ne
+  //
+  //  ->
+  //
+  // ctest(X, X, cflags/~cflags, cc0/~cc0, flag0)
+  // brcond cc1
+  //
+  //
+  // sub(and/or(setcc(cc0, flag0), setcc(cc1, cmp (X, 0))), 1)
+  // brcond ne
+  //
+  //  ->
+  //
+  // ctest(X, X, cflags/~cflags, cc0/~cc0, flag0)
+  // brcond ~cc1
```

Comment:
Can you make these comments a bit more compact? How about:
```
  // cmp(and/or(setcc(cc0, flag0), setcc(cc1, sub/cmp (X, Y))), 0)
  // brcond ne
  //  ->
  //    ccmp(X, Y, cflags/~cflags, cc0/~cc0, flag0)
  //    brcond cc1
  //
  // sub(and/or(setcc(cc0, flag0), setcc(cc1, sub/cmp (X, Y))), 1)
  // brcond ne
  //  ->
  //    ccmp(X, Y, cflags/~cflags, cc0/~cc0, flag0)
  //    brcond cc1
```

Reply:
Done